File: src/main/java/com/schoolproject/airbnbclone/models/Listing.java
package com.schoolproject.airbnbclone.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Listing {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false)
    private Long id;

    @Column(nullable = false)
    private String name;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "location_id", nullable = false)
    private Location location;

    @OneToMany(mappedBy = "listing")
    private List<Image> images = new ArrayList<>();

    @OneToMany(mappedBy = "listing")
    private List<Availability> availabilities = new ArrayList<>();

    @ManyToOne
    @JoinColumn(name = "host_id", referencedColumnName = "id")
    private User host;

    @Column(nullable = false)
    private Integer maxPeople;

    @Column(nullable = false)
    private BigDecimal price;

    @Column(nullable = false)
    private Integer extraPeople;

    @Column(nullable = false)
    private String type;

    @Column(nullable = false)
    private Integer beds;

    @Column(nullable = false)
    private Integer baths;

    @Column(nullable = false)
    private Integer meters;

    @Column(nullable = false)
    private Boolean living;

    @Column(nullable = false)
    private Boolean party;

    @Column(nullable = false)
    private Boolean pets;

    @Column(nullable = false)
    private String summary;

    @Column(nullable = false)
    private Boolean wifi;

    @Column(nullable = false)
    private Boolean ac;

    @Column(nullable = false)
    private Boolean heat;

    @Column(nullable = false)
    private Boolean kitchen;

    @Column(nullable = false)
    private Boolean tv;

    @Column(nullable = false)
    private Boolean parking;

    @Column(nullable = false)
    private Boolean elevator;
}

File: src/models/Listing/Listing.ts
import { ILocation } from "../Location";

export interface IListing {
    location: ILocation;
    name: string;
    startDate: string;
    endDate: string;
    maxPeople: number;
    price: number;
    extraPeople: number;
    type: string;
    beds: number;
    baths: number;
    meters: number;
    living: boolean;
    party: boolean;
    pets: boolean;
    summary: string;
    wifi: boolean;
    ac: boolean;
    heat: boolean;
    kitchen: boolean;
    tv: boolean;
    parking: boolean;
    elevator: boolean;
}

export class Listing implements IListing {
    location: ILocation;
    name: string;
    startDate: string;
    endDate: string;
    maxPeople: number;
    price: number;
    extraPeople: number;
    type: string;
    beds: number;
    baths: number;
    meters: number;
    living: boolean;
    party: boolean;
    pets: boolean;
    summary: string;
    wifi: boolean;
    ac: boolean;
    heat: boolean;
    kitchen: boolean;
    tv: boolean;
    parking: boolean;
    elevator: boolean;

    constructor(
        location: ILocation,
        name: string,
        startDate: string,
        endDate: string,
        maxPeople: number,
        price: number,
        extraPeople: number,
        type: string,
        beds: number,
        baths: number,
        meters: number,
        living: boolean,
        party: boolean,
        pets: boolean,
        summary: string,
        wifi: boolean,
        ac: boolean,
        heat: boolean,
        kitchen: boolean,
        tv: boolean,
        parking: boolean,
        elevator: boolean
    ) {
        this.location = location;
        this.name = name;
        this.startDate = startDate;
        this.endDate = endDate;
        this.maxPeople = maxPeople;
        this.price = price;
        this.extraPeople = extraPeople;
        this.type = type;
        this.beds = beds;
        this.baths = baths;
        this.meters = meters;
        this.living = living;
        this.party = party;
        this.pets = pets;
        this.summary = summary;
        this.wifi = wifi;
        this.ac = ac;
        this.heat = heat;
        this.kitchen = kitchen;
        this.tv = tv;
        this.parking = parking;
        this.elevator = elevator;
    }
}